### PR TITLE
Remove IsOperationAllowed() method from Authorizer interface

### PR DESCRIPTION
**Pull request by jeyhun**
Removed `IsOperationAllowed()` method from `Authorizer` interface. Moved its functionality to _database.go_

### CODE REVIEW

File: data-access-layer/datastore/database.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (a DbRoleSlice) Len() int {
 	return len(a)
 }
 
-func (dbRole DbRole) IsTenantDbRole() bool {
+func (dbRole DbRole) isDbRoleTenantSpecific() bool {
```
Comment (krishnamiriyala) [inline data-access-layer/datastore/database.go:58]: IsDbRoleTenantScoped()

File: data-access-layer/datastore/database.go [hunk 2]
```diff
@@ -281,17 +281,26 @@ func getTx(db *sql.DB, orgId string, isMultitenant bool) (*sql.Tx, error) {
...
-		logger.Errorf("Operation not allowed for the following reason: %s", err.Error())
-		return nil, err
+	/*
+		If the DB role is tenant-specific (TENANT_READER or TENANT_WRITER) and the table is multi-tenant,
+		make sure that the record being inserted/modified/updated/deleted/queried belongs to the user's org.
+		If operation is SELECT but no specific tenant's data is being queried (e.g., FindAll() was called), allow the operation to proceed.
+	*/
+	if dbRole.isDbRoleTenantSpecific() && IsMultitenant(record, tableName) {
+		orgIdCol := getField(ORG_ID_COLUMN_NAME, record)
+		if orgIdCol != "" && orgIdCol != orgId {
```
Comment (krishnamiriyala) [inline data-access-layer/datastore/database.go:296]: should we fail here for non SELECT operations here based on line #292?

**Review by krishnamiriyala — CHANGES_REQUESTED** [review]

**Comment by krishnamiriyala** [comment]
@jeyhun Can you also have PR title in sync with commit message something like `Removed IsOperationAllowed() method from Authorizer interface`

File: data-access-layer/datastore/database.go (hunk 2 above)
Comment (jeyhun) [reply data-access-layer/datastore/database.go:296]: INSERT, UPDATE, DELETE will fail if you're trying to insert/update/delete a record of another tenant.
SELECT will also fail if you're trying to read a single record of another tenant.
Only if SELECT is trying to read all tenants' data will this if statement have no effect.

**Comment by jeyhun** [comment]
> @jeyhun Can you also have PR title in sync with commit message something like `Removed IsOperationAllowed() method from Authorizer interface`

Done

File: data-access-layer/datastore/database.go (hunk 2 above)
Comment (jeyhun) [reply data-access-layer/datastore/database.go:296]: so, non-SELECT operations are already failing if you're trying to modifying another tenant's data

File: data-access-layer/datastore/database.go (hunk 1 above)
Comment (jeyhun) [reply data-access-layer/datastore/database.go]: Renamed the method. Also renamed the booleans that take the value returned by this method to `IsDbRoleTenantScoped`

**Review by krishnamiriyala — APPROVED** [review]

**Pull request merged**